### PR TITLE
SAMD51 LCD support

### PR DESCRIPTION
**Pull request by GMagician**
First things to compile with lcd support

**needs https://github.com/MarlinFirmware/U8glib-HAL/pull/15 or 16 to be merged before**

### CODE REVIEW

File: Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 #include "../../inc/MarlinConfig.h"
 
-#if ENABLED(U8GLIB_ST7920) && !defined(U8G_HAL_LINKS) && !defined(__SAM3X8E__)
+#if ENABLED(U8GLIB_ST7920) && !defined(U8G_HAL_LINKS) && !defined(__SAM3X8E__) && !defined(__SAMD51__)
```
Comment (tpruvot) [inline Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp]: shouldnt this be if ``defined(__AVR__)`` ?

Comment (GMagician) [reply Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp]: This class seems incompatible with samd51 like it's for due, don't know for other boards

Comment (thinkyhead) [reply Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp]: > shouldnt this be if defined(__AVR__) ?

It looks like the original author intended more than AVR, but we could try it.

Comment (thinkyhead) [reply Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp]: Apparently Teensy 3.5/3.6 wants this too.

Comment (thinkyhead) [reply Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp]: Actually, it wants the alternative. Patched.

**Comment by tpruvot** [comment]
i add a comment there to ping @thinkyhead related to the SKRpro (F4) and the mini (F1) issues, unsure the issues links do a notify.

so it look like we need to replace `defined(__AVR__)` by `ANY(__AVR__, ARDUINO_ARCH_STM32)` in the 2 dogm cpp files...

but maybe its even more platforms.... only seen reports from these 2 boards for now...

**Pull request merged**